operator/k8s_cep_gc.go

```diff
@@ -96,6 +98,10 @@ func enableCiliumEndpointSyncGC() {
 
 					// For each CEP we fetched, check if we know about it
 					for _, cep := range ceps.Items {
+						if cep.Status.Identity != nil {
+							identityHeartbeat.MarkAlive(strconv.FormatInt(cep.Status.Identity.ID, 10), timeNow)
```

Comment:
Can't this logic be in the CEP watcher? With a reference counter we would be able to know which CEPs are using a particular identity.

Reply:
Which CEP watcher?

Reply:
Apparently a non-existing one. I was thinking on having a CEP watcher in the operator that would keep a reference counter for the used identities.

Reply:
The watcher has a downside though. If we ever miss the notification, we won't ever clean up again unless the identity gets re-used and released again. We are also no longer talking about a heartbeat-based mechanism but event notification with the usual pros and cons.